### PR TITLE
Feature/add grab strategy to start grabbing

### PR DESCRIPTION
**Pull request by Vincent-Beltman-Inspiro**

### CODE REVIEW

File: src/pylon-cxx-rs.cc [hunk 1]
```diff
@@ -6,11 +6,11 @@
 #include "pylon/PylonIncludes.h"
 #include "pylon-cxx-rs.h"
 
-std::unique_ptr<std::vector<std::string>> to_std_vec_str(Pylon::StringList_t names)
+std::unique_ptr<std::vector<std::string>> to_std_vec_str(const Pylon::StringList_t& names)
```
Comment (Vincent-Beltman-Inspiro) [inline src/pylon-cxx-rs.cc:9]: Unrelated change. Let me know if this is ok.

**Review by astraw — COMMENTED** [review]
Can you add to impl GrabOptions:

```rust
    pub fn strategy(self, strategy: GrabStrategy) -> GrabOptions {
        Self {
            count: self.count,
            strategy: Some(strategy),
        }
    }
```

File: src/lib.rs [hunk 2]
```diff
@@ -390,11 +408,12 @@ impl<'map, 'parent: 'map> NodeMap<'map, 'parent> {
 #[derive(Default)]
 pub struct GrabOptions {
     count: Option<u32>,
+    strategy: Option<GrabStrategy>,
 }
 
 impl GrabOptions {
     pub fn count(self, count: u32) -> GrabOptions {
-        Self { count: Some(count) }
+        Self { count: Some(count), strategy: None }
```
Comment (astraw) [inline src/lib.rs:416]: This should be `strategy: self.strategy`

**Comment by astraw** [comment]
Thanks! This looks OK to me. Can you run 'cargo fmt' on a final commit?

**Comment by Vincent-Beltman-Inspiro** [comment]
@astraw Comments are resolved

**Comment by astraw** [comment]
Many thanks. I made a minor change to preserve the signature of `GrabOptions::count` unchanged and then make `GrabOptions::strategy` consistent. (I'm not sure why `cargo semver-checks` didn't complain about your version, but I thought it was better to be on the safe side and preserve the signature as-is.)

Released as 0.4.2.
